components/prism-ini.js

```diff
@@ -1,11 +1,37 @@
 Prism.languages.ini= {
-	'comment': /^[ \t]*[;#].*$/m,
-	'selector': /^[ \t]*\[.*?\]/m,
-	'constant': /^[ \t]*[^\s=]+?(?=[ \t]*=)/m,
-	'attr-value': {
-		pattern: /=.*/,
+	'comment': {
+		pattern: /(^[ \f\t\v]*)[#;][^\n\r]*/m,
+		lookbehind: true
+	},
+	'header': {
+		pattern: /(^[ \f\t\v]*)\[[^\n\r\]]*\]?/m,
+		lookbehind: true,
 		inside: {
-			'punctuation': /^[=]/
+			'section-name': {
+				pattern: /(\[[ \f\t\v]*)[^ \f\n\r\t\v\]]+(?:[ \f\t\v]+[^ \f\n\r\t\v\]]+)*/,
```

Comment:
1. The `header` pattern already assures us `[^\n\r\]]`, so we don't have exclude `\r\n` here.
2. A `^` at the start of the pattern isn't necessary here for correctness but it allows the regex engine to better optimize the pattern.

```suggestion
				pattern: /(^\[[ \f\t\v]*)[^ \f\t\v\]]+(?:[ \f\t\v]+[^ \f\t\v\]]+)*/,
```

Reply:
1. This is a bright idea.
2. This is another bright idea.